Makefile

```diff
@@ -59,3 +70,5 @@ test_rehash:
 scratch:
 	cargo watch --exec "run --release --bin scratch"
 
+application/node_modules: application/yarn.lock
+	cd application && yarn install
```

Comment:
The idea is that this block provides make with instructions for producing the node modules directory. Then when you give `application/node_modules` as a dependency to another target make knows that it needs to run `yarn install` to get that directory; and it knows that in order for node modules to be considered up-to-date it should run `yarn install` again if `application/yarn.lock` has changed.